### PR TITLE
E2E: move catalog source to 4.15

### PR DESCRIPTION
**Pull request by alfonsomthd**

### CODE REVIEW

File: openshift-ci/odf-catalog-source.yaml [hunk 1]
```diff
@@ -62,7 +62,7 @@ spec:
       type: RuntimeDefault
   containers:
     - name: rhceph-dev-icsp
-      image: quay.io/rhceph-dev/ocs-registry:latest-stable-4.14.0
+      image: quay.io/rhceph-dev/ocs-registry:latest-stable-4.15.0
```
Comment (SanjalKatiyar) [inline openshift-ci/odf-catalog-source.yaml:65]: even though it is tagged as `4.15` this image will be based on current `master` right (during dev phase) ?? and not `release-4.15` branch ??

Comment (SanjalKatiyar) [reply openshift-ci/odf-catalog-source.yaml:65]: can we please confirm this once with someone from build team ??

Comment (SanjalKatiyar) [reply openshift-ci/odf-catalog-source.yaml:65]: just good to know... not blocking the PR...

Comment (alfonsomthd) [reply openshift-ci/odf-catalog-source.yaml:65]: @SanjalKatiyar I confirmed with Boris that for now `4.15` builds are based on `master/main` branches.

**Pull request merged**